combat-trainer.lic

```diff
@@ -197,12 +198,16 @@ class SetupProcess
 
   def determine_next_to_train(game_state, weapon_training, ending_ranged)
     return unless game_state.skill_done? || !weapon_training[game_state.weapon_skill] || ending_ranged
-    #check if all weapons are maxed exp to prevent rapid cycling
-    if weapon_training.reject { | skill,_ |  DRSkill.getxp(skill) == 34 }.empty?
-      echo 'all weapons locked, not switching' if $debug_mode_ct
-      return
+    
+    unless @ignore_weapon_mindstate
+      #check if all weapons are maxed exp to prevent rapid cycling
+      if weapon_training.reject { | skill,_ |  DRSkill.getxp(skill) == 34 }.empty?
+        echo 'all weapons locked, not switching' if $debug_mode_ct
+        return
+      end
     end
```

Comment:
Some suggestions (any of these can be done or not done independently):
1. Extract the `reject` call into a named var for clarity
2. Use `all?` instead of `reject {...}.empty?`
3. Merge the conditionals

```suggestion
    #check if all weapons are maxed exp to prevent rapid cycling
    no_weapons_to_train = weapon_training.all? { | skill,_ |  DRSkill.getxp(skill) == 34 }
    if no_weapons_to_train && !@ignore_weapon_mindstate
        echo 'all weapons locked, not switching' if $debug_mode_ct
        return
    end
```

Reply:
Initially it didn't make sense to me, but I added echoes, and tested, and this works well. Thank you for the suggestion.